### PR TITLE
Hide Banned Projects from the Site

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/readthedocs/readthedocs.org/issues/5152

**Can View Banned Projects:** Project User, Admin User
**Can not View Banned Projects:** Anonymous User, Regular User

### CODE REVIEW

**Comment by saadmk11** [comment]
It seems like if the project has no user or the project users have no profile then the project will not be visible by Anonymous User, Regular User. That's why tests are failing as many tests don't meet those requirements. @readthedocs/core Is this (not showing these kinds of projects) something we want?

**Comment by humitos** [comment]
Hrm, this is a good question.

The goal of this PR is to hide projects from banned users. The only way to know the user is banned is from their profile. IMO, if the user does not have profile or the project does not have users, it should be returned in the queryset. Otherwise, we will be excluding projects that we are not 100% sure they are banned.

**Comment by saadmk11** [comment]
@humitos Sounds good to me. I have updated the PR with a bit different approach so that we don't filter out those cases. Please have a look :)

File: readthedocs/projects/querysets.py [hunk 1]
```diff
@@ -37,26 +37,32 @@ def for_admin_user(self, user):
 
     def public(self, user=None):
         queryset = self.filter(
-            privacy_level=constants.PUBLIC,
-            users__profile__banned=False
+            Q(users__profile__banned=False) |
+            Q(users__profile__isnull=True) |
+            Q(users__isnull=True),
```
[the remark this answers is not in the document]
Comment (saadmk11) [reply readthedocs/projects/querysets.py]: Done! :)

**Comment by saadmk11** [comment]
> I'm -1 on this change. It will make our queries much slower. We already removed this logic from the homepage view because it caused downtime from being slow, and this change seems like it will just make _all_ of our Project queries much slower.
> 
> We should just delete these projects, not make our queries way more complex.
> 
> Sorry for the confusion here, but I didn't quite realize how slow these queries were, but we definitely shouldn't be introducing them on all Project queries.

Where do you suggest adding these filters? Or should we just stick with deleting banned projects and do the queries as before?

**Comment by stsewd** [comment]
So, we already have a semi-automatic way of deleting spam projects, so I guess we don't need to filter spam projects anymore (we hit an issue when we tried something like this, the queries are very slow)

**Comment by ericholscher** [comment]
Sorry for not replying here -- I think we should just skip this logic, and delete projects that are banned. We have now automated the deletion, so once they're banned they should be deleted. I think we can close this.

**Pull request closed without merging**